CRM/Contribute/Page/UserDashboard.php

```diff
@@ -60,11 +60,23 @@ public function listContribution() {
     // We want oldest first, just among the most recent contributions
     $rows = array_reverse($rows);
 
-    foreach ($rows as $index => $row) {
+    foreach ($rows as $index => &$row) {
       // This is required for tpl logic. We should move away from hard-code this to adding an array of actions to the row
       // which the tpl can iterate through - this should allow us to cope with competing attempts to add new buttons
       // and allow extensions to assign new ones through the pageRun hook
-      $row[0]['contribution_status_name'] = CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $row['contribution_status_id']);;
+      if ('Pending' === CRM_Core_PseudoConstant::getName('CRM_Contribute_BAO_Contribution', 'contribution_status_id', $row['contribution_status_id'])) {
```

Comment:
Hope to see "Partially Paid" here soon. I know work is in progress for that. Thank you for that Eileen!